### PR TITLE
Remove components

### PR DESCRIPTION
**Pull request by tristen**
Drop components as deemed underutilized or no longer needed per https://github.com/mapbox/mr-ui/issues/179

### CODE REVIEW

File: src/components/control-radio-set/control-radio-set.js [hunk 1]
```diff
@@ -1,7 +1,6 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import omit from '../utils/omit';
-import ControlLegend from '../control-legend';
```
Comment (tristen) [inline src/components/control-radio-set/control-radio-set.js:4]: Some of our control group elements optionally used `ControlLegend` so they have been dropped from these maintained components along with the supported props related to it.

**Review by VLuisa — COMMENTED** [review]
Overall this looks good!

@tristen I don't feel super strongly about this but just wanted to say I’m not entirely convinced we should get rid of ‘Badge’ - we do use the [badge pattern](https://github.com/mapbox/studio/blob/9ac1fa9e2b8f998f6dca86a7795c2ad0c2efb460/src/shared_components/announcement_item_tag.js) in Studio, we just haven’t made use of mr-ui for it and I do think it could be much cleaner to have the component than having to manually recreate all the css classes for every tag color combination (correct me if that’s not what we’d have to do). 

image

**Comment by tristen** [comment]
oh yeah @VLuisa that's a good point to raise. I could be convinced to keep it. I was thinking the CSS classes used to create each badge could be added to https://github.com/mapbox/studio/blob/main/src/util/ui_util/get_style.ts which is a more shareable place within Studio to maintain consistency.

**Comment by VLuisa** [comment]
@tristen hmm i think i actually like that more. And if it's in Studio then we coulddd be more opinionated than in mr-ui and use a semantic naming structure (ex: `badgeNew`, `badgeBeta`, etc), kind of like they're set up in this tag component... 
```
'bg-orange-faint color-orange': type === 'deprecated',
'bg-green-faint color-green': type === 'new',
'bg-blue-faint color-blue': type === 'beta',
'bg-pink-faint color-pink': type === 'ui'
```

**Pull request merged**